### PR TITLE
Refactor script and implement the specialized interface

### PR DESCRIPTION
**Pull request by abelsiqueira**
This refactor #26 squashing similar commits.
The work could be divided into three parts:
  - Refactoring
  - Creating the specialized interface
  - Creating the tests for the specialized interface

### CODE REVIEW

File: src/specialized_interface.jl [hunk 1]
```diff
@@ -0,0 +1,3151 @@
...
+    $(Cint[io_buffer]), $(Cint[n]), $(x), $(x_l), $(x_u), $(libname))
+  @cutest_error
+  return x, x_l, x_u
+end
+
+function usetup!(input::Int, out::Int, io_buffer::Int, n::Int, x::Array{Float64, 1},
+    x_l::Array{Float64, 1}, x_u::Array{Float64, 1}, libname)
+  io_err = Cint[0]
+  @eval usetup($(io_err), $(Cint[input]), $(Cint[out]),
+    $(Cint[io_buffer]), $(Cint[n]), $(x), $(x_l), $(x_u), $(libname))
```
Comment (dpo) [inline src/specialized_interface.jl]: Shouldn't the `n` argument have a `&` in front of it, as your recent update to the Julia interface?

**Comment by dpo** [comment]
Would you mind also updating `runtests_unc.jl` so the tests run on HS4?

**Comment by abelsiqueira** [comment]
Unconstrained tests updated.

**Comment by dpo** [comment]
Thanks!

**Pull request merged**